test/statestoJSON.test.js

```diff
@@ -0,0 +1,8 @@
+const factory = require('../factory.js');
+const JSONMap = require('../toJSONMap.js');
+
+Object.entries(JSONMap).forEach(([dataType, stateJSON]) => {
+  test('toJSON method returns an correct representation of a toJSON string for the specified class', () => {
+    expect(factory(dataType).toJSON()).toMatchObject(stateJSON);
```

Comment:
```suggestion
    expect(factory(dataType).toJSON()).toEqual(stateJSON);
```

I think we want to use `toEqual` here since it will be more precise and do a deep equality check on all properties of the objects.